src/libsyntax/parse/parser/stmt.rs

```diff
@@ -41,11 +42,29 @@ impl<'a> Parser<'a> {
         let lo = self.token.span;
 
         Ok(Some(if self.eat_keyword(kw::Let) {
```

Comment:
It may be a good idea to put all of the "parse let binding"-related code into one function since they share some code, e.g. `mk_dummy_stmt`, which could then be inlined.